packages/grid/_modules_/grid/components/cell/GridCell.tsx

```diff
@@ -128,7 +128,11 @@ export const GridCell: React.FC<GridCellProps> = React.memo((props) => {
   };
 
   React.useLayoutEffect(() => {
-    if (hasFocus && cellMode === 'view' && cellRef.current) {
+    if (
+      hasFocus &&
+      cellRef.current &&
+      (!document.activeElement || !cellRef.current!.contains(document.activeElement))
+    ) {
       cellRef.current!.focus();
     }
   });
```

Comment:
I would personally remove the whole block for Point 5. of https://github.com/mui-org/material-ui-x/issues/1300 and #1295.

Reply:
it's fixed